src/cloudWatchLogs/changeLogSearch.ts

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as telemetry from '../shared/telemetry/telemetry'
+import * as vscode from 'vscode'
+import { showInputBox } from '../shared/ui/inputPrompter'
+import { isLogStream } from './cloudWatchLogsUtils'
+import { prepareDocument } from './commands/searchLogGroup'
+import { getActiveUri } from './document/logStreamDocumentProvider'
+import { CloudWatchLogsData, filterLogEventsFromUriComponents, LogStreamRegistry } from './registry/logStreamRegistry'
+import { isViewAllEvents, TimeFilterResponse, TimeFilterSubmenu } from './timeFilterSubmenu'
+
+type ChangeableParam = 'filterPattern' | 'timeFilter'
+
+export async function getNewData(
+    registry: LogStreamRegistry,
+    param: ChangeableParam,
+    oldUri: vscode.Uri
+): Promise<CloudWatchLogsData | undefined> {
+    const oldData = registry.getLogData(oldUri) as CloudWatchLogsData
```

Comment:
Is there every a time where as is appropriate? Other than maybe in some testing scenarios where I've seen it used.

Reply:
> Is there ever a time where as is appropriate

Yeah there's plenty of valid uses. It's a shortcut for when a high-level of precision is not needed. But it's still a shortcut. And lots of shortcuts cause problems.

Using `as` is fine as long as we _know_ that the type assertion is valid in whatever context we're working in. By that I mean the assertion will not cause errors later on. Still best to avoid it though because, by design, it causes a "break" in the type system. Type assertions are a powerful tool if used wisely.